Plugins/Events/Events/CombatEvents.cpp

```diff
@@ -42,4 +63,56 @@ void CombatEvents::StartCombatRoundHook(
     Events::SignalEvent(before ? "NWNX_ON_START_COMBAT_ROUND_BEFORE" : "NWNX_ON_START_COMBAT_ROUND_AFTER" , thisPtr->m_pBaseCreature->m_idSelf);
 }
 
+void CombatEvents::AttackHook(
+    Hooks::CallType type,
+    API::CNWSCombatRound* thisPtr,
+    unsigned char attackNumber)
+{
+    if ( type == Hooks::CallType::AFTER_ORIGINAL )
+    {
+        // SetCurrentAttack is 1-based, GetAttack is 0-based
+        CNWSCombatAttackData* combatAttackData = g_combatEvents->m_combatAttackData = thisPtr->GetAttack(attackNumber - 1);
+        Events::PushEventData("TARGET_OBJECT_ID", Utils::ObjectIDToString(thisPtr->m_pBaseCreature->m_oidAttackTarget));
+        Events::PushEventData("ATTACK_NUMBER", std::to_string(static_cast<int>(attackNumber)));
+        for ( int i = 0; i < 13; i++ )
+        {
+            int16_t nDamage = combatAttackData->m_nDamage[i];
+            Events::PushEventData("ATTACK_DAMAGE_" + std::to_string(i), std::to_string(nDamage));
+        }
+        Events::PushEventData("ATTACK_RESULT", std::to_string(combatAttackData->m_nAttackResult));
+        Events::PushEventData("IS_SNEAK_ATTACK", std::to_string(combatAttackData->m_bSneakAttack || combatAttackData->m_bDeathAttack));
+        Events::PushEventData("IS_OFFHAND_ATTACK", std::to_string(combatAttackData->m_nWeaponAttackType == 2));
+        Events::SignalEvent("NWNX_ON_ATTACK", thisPtr->m_pBaseCreature->m_idSelf);
```

Comment:
We generally signal a *_BEFORE and *_AFTER event, see the other events for an example.

Reply:
Didn't do this since there doesn't seem to be any notable difference between the two - in both cases the script would run after damage has been calculated but before it is being dealt.